plugins/out_stackdriver/stackdriver.c

```diff
@@ -277,6 +277,114 @@ static char *get_google_token(struct flb_stackdriver *ctx)
     return ctx->o->access_token;
 }
 
+static int process_local_resource_id(const void *data, size_t bytes,
+                                     struct flb_stackdriver *ctx, char *type)
+{
+    int i;
+    int len;
+    int ret_code = -1;
+    int flag = 0;
+    size_t off = 0;
+    msgpack_object k;
+    msgpack_object v;
+    msgpack_object root;
+    msgpack_object_map map;
+    msgpack_unpacked result;
+    char *local_resource_id;
+    char *ptr;
+    const char delim[] = ".";
+
+    msgpack_unpacked_init(&result);
+    while (msgpack_unpack_next(&result, data, bytes, &off) == MSGPACK_UNPACK_SUCCESS && !flag) {
+        root = result.data;
+        flag = 0;
+
+        if (root.type != MSGPACK_OBJECT_ARRAY ||
+            root.via.array.size != 2 ||
+            root.via.array.ptr[1].type != MSGPACK_OBJECT_MAP) {
+            flb_plg_warn(ctx->ins, "unexpected record format");
+            continue;
+        }
+
+        map = root.via.array.ptr[1].via.map;
+        len = strlen(LOCAL_RESOURCE_ID_KEY);
+        for (i = 0; i < map.size; i++) {
+            k = map.ptr[i].key;
+            v = map.ptr[i].val;
+
+            if (k.type == MSGPACK_OBJECT_STR && 
+                strncmp(k.via.str.ptr, LOCAL_RESOURCE_ID_KEY, len) == 0) {
+                
+                ret_code = 0;
```

Comment:
Is ret_code useful here?
This if condition would be met once, right?

Reply:
deleted. Put logic in the following if conditions.